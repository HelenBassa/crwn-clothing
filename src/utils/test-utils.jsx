import { render } from '@testing-library/react';
import { BrowserRouter as Router } from "react-router-dom";
import { Provider } from "react-redux";
import { store } from "../store/store";


afterAll(() => jest.clearAllMocks());

const AllTheProviders = ({ children }) => {
  return (
    <Provider store={store}>
      <Router>
        {children}
      </Router>
    </Provider>
  );
};

const customRender = (ui, options) => render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react';
export * from '@testing-library/jest-dom';
export * from '@testing-library/user-event';

export { customRender as render };