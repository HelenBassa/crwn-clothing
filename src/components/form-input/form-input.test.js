import { render } from "../../utils/test-utils";
import FormInput from "./form-input.component";

it("should render FormInput component", () => {
  const mockLabel = "";

  const { container } = render(<FormInput label={mockLabel} />);
  expect(container).toMatchSnapshot();
});
