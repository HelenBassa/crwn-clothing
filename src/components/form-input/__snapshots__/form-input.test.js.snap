// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render FormInput component 1`] = `
<div>
  <div
    class="sc-dkrFOg fSosQj"
  >
    <input
      class="sc-gswNZR hmZPfH"
    />
  </div>
</div>
`;
